### ISSUE TITLE
Mascope UX improvements

### ISSUE THREAD

**Issue by oskarsari** [ultra-trace-systems/mascope#348]
- [x] In acquisitions tab, show number of selected acquisitions
- [x] In batch import dialog, show instructions what to paste
- [x] Process batch: pass mz calibration parameters to the endpoint
- [x] Export batch/peaks works only if batch selected, otherwise fails silently (except console log)
- [x] Log notifications, to be able to see recently closed notifications
- [x] Recalibrate batch UI feature
- [x] Remove stale progress bars
- [x] Progress bar tooltip always says `"Computing sample batch matches, processing sample **1**/n"`
- [ ] ~In sample browser, indicate batch/item processing~ _see comment below_
- [ ] ~Stack progress bars vertically?~ _see comment below_

**Comment by philip-chernonog** [comment]
- [x] The match score tag of the sample shouldn't be displayed if sample “matched” property is 0. Maybe to add this as a first check to sample match score tag displaying

**Comment by ixxie** [comment]
> * [ ]  The match score tag of the sample shouldn't be displayed if sample “matched” property is 0. Maybe to add this as a first check to sample match score tag displaying

This is something which was explicitly requested by @oskarsari, if I recall correctly.

**Comment by ixxie** [comment]
### Progress Bar Complexity

Some of the items mention progress bars. Some issues I can and will resolve immediately, but some issues are running a bit deeper. These relate to **complex processes** composed of multiple subprocesses related through parent-child relationships.

Complex processes result in **too many progress bars**, and since these are stacked in the z-index, it becomes hard to follow. Vertical stacking is not a good idea, because we have no strict upper limit to the number of progress bars, and the result would anyway be visually confusing.

Additionally, complex processes can result in an **incorrect cascading of failures**. If a child, parent or sibling process fails, should the process also be considered as failed? This is not obvious or easy to answer in a general way.

### Proposed Solution

The need to think about complex nested process is legitimate, but perhaps we can simplify the API with the following principles:

**1. Hide child processes from the frontend** - when the user requests a complex task (e.g. import batch) which requires multiple subprocesses, it suffices to inform them of the *overall* progress, and hide subprocess progress notifications altogether. Child progress notifications should never be emitted to the frontend.

**2. Parent processes dictate failure conditions in the backend** - cascade of failures is dependent on the context of the particular process at hand, so the backend endpoint running it should decide what happens if a child process fails. 

**3. Expose subprocess details in progress messages** - if we want to inform the user of progress of subprocesses, we can still amend the message sent in progress notifications of the parent, e.g. `calibration completed, matching ongoing`. These could be should in the tooltip of the progress bar. If we want to go a step further, we could formalize an API describing a process **phase**, that would allow us to do things like change the color of the progress bar.

I hope the result will be easier for users and developers to understand and work with.

**Comment by ixxie** [comment]
### Batch & Item Processing Indication

When I considered this, my first question was: what does 'processing' mean. Likely, it would be any of the following processes: _rematching_, _recalibrating_ and _importing_ new samples.

With the current notification structure, retrieving this information is not particularly easy. Its not impossible either, but to do it well would require at least half a days work, to my estimation.

### Locks

This topic led me to think about a related topic: **locks**. If we would want to prevent users from overriding each-others work accidently, it would be helpful to indicate to the application that certain scopes are locked, and show the user that they cannot perform certain actions if certain locks are in place.

Incidentally, this is precisely when some process is ongoing. Such an API would be relatively simple: emit a `lock` event which includes a `type` field (e.g. `sample_batch` lock) and a `payload` field (e.g. `{sample_item_id: 'xyz123'}`). A similar `unlock` event would also be implemented.

The bulk of the work involved would be in identifying the right places to apply and - possibly - persist lock state in the backend and frontend. However, this work can be undertaken incrementally, e.g implementing batch locks first and then proceeding to other locks.

If this is something we are interested in implementing in the foreseeable future, I believe we are better off implementing the 'processing' indicator after implementing locks.

### Invalidation

My gut says there is some analogy here with the concept of data invalidation. I'm not sure if the analogy is strong enough to make it prudent to implement these two functionality at once, but I'd like to explore the possibility together.

**Comment by oskarsari** [comment]
> > * [x]   The match score tag of the sample shouldn't be displayed if sample “matched” property is 0. Maybe to add this as a first check to sample match score tag displaying
> 
> This is something which was explicitly requested by @oskarsari, if I recall correctly.

There is a difference. Sample can be _matched_ with match score 0%. This is what I requested to show in the UI. What @philip-chernonog is referring to, we should not show match score 0% if the sample is _not matched_, meaning the matches have not been computed (we don't know if match score is 0 or something else). There is a property `matched` (0/1) to indicate this status

**Comment by oskarsari** [comment]
> * [x]  The match score tag of the sample shouldn't be displayed if sample “matched” property is 0. Maybe to add this as a first check to sample match score tag displaying

We agreed with @ixxie that he will fix the quoted item and make issues about the last two remaining: progress bars and process indicators, due to their complexity.

**Comment by philip-chernonog** [comment]
- [x] Order of the batches by creation time sample_batch_utc_created
- [x] consistent Warnings  Invalid prop: type check failed for prop "onLabel". Expected String with value "[object Object]", got Object  
![Untitled](https://github.com/Karsa-Oy/Mascope/assets/162338346/43d8f11f-ffe0-4486-ba4a-8980588d09e7)
- [x] Increase retentionLimit  to 200 of notifications log and add the search bar 
- [x] Not able to edit the sample item type if generate filter id before that in some cases (scroll and dropdown issue)
- [x] Match score flickering to 0 in the target browser when selecting between samples (target browser/sample matched property)
- [x] Reversed order of the acquisitions and sample items during the sample items import
- [ ] ~Is it possible to update progress tooltip if keep courser on it?~ nope :(
